'ready' function for client services clashes with 'ready' rpc method

## Bug Report

### Version
├── tonic v0.1.0-alpha.6
└── tonic-build v0.1.0-alpha.6

### Crates
tonic-build

### Description

I started looking at using tonic with [this Agones SDK protocol](https://github.com/googleforgames/agones/blob/master/sdk.proto) today and found that the generated code wouldn't compile because there were multiple 'ready' functions for the service. (ref: https://github.com/hyperium/tonic/blob/4471a5f19ce9d88583605be1b58f119a2775a7e6/tonic-build/src/client.rs#L33)

I guess it's going to be relatively likely for "ready" to conflict with more than just this protocol, so I wonder if it's possible for this to be changed to a less common name?

Yep, here's the original compilation failure I hit:
```bash
error[E0201]: duplicate definitions with name `ready`:
   --> /home/rib/src/realfit-server/target/debug/build/realfit-server-808a11d4a9ed943b/out/agones.dev.sdk.rs:133:9
    |
124 | /         pub async fn ready(&mut self) -> Result<(), tonic::Status> {
125 | |             self.inner.ready().await.map_err(|e| {
126 | |                 tonic::Status::new(
127 | |                     tonic::Code::Unknown,
...   |
130 | |             })
131 | |         }
    | |_________- previous definition of `ready` here
132 |           #[doc = " Call when the GameServer is ready"]
133 | /         pub async fn ready(
134 | |             &mut self,
135 | |             request: impl tonic::IntoRequest<super::Empty>,
136 | |         ) -> Result<tonic::Response<super::Empty>, tonic::Status> {
...   |
140 | |             self.inner.unary(request.into_request(), path, codec).await
141 | |         }
    | |_________^ duplicate definition

error: aborting due to previous error

For more information about this error, try `rustc --explain E0201`.
error: could not compile `realfit-server`.
```

So, the issue is that there can be an rpc method named 'ready' which tonic isn't in control of, which can be in addition to the one generated by tonic.

This is the workaround I'm currently using, which fixes the issue for me:
https://github.com/rib/tonic/commit/ee60bc702d8c4bef3a5f59354d3aa8ea3bb7c67d
(though I didn't give the alternate naming any real thought, or consider if there's maybe a different way of structuring things to better avoid the possibility of clashing)

--- a/tonic-build/src/client.rs
+++ b/tonic-build/src/client.rs
@@ -34,11 +34,14 @@ pub(crate) fn generate(service: &Service, proto: &str) -> TokenStream {
                     Self { inner }
                 }
 
-                /// Check if the service is ready.
-                pub async fn ready(&mut self) -> Result<(), tonic::Status> {
-                    self.inner.ready().await.map_err(|e| {
-                        tonic::Status::new(tonic::Code::Unknown, format!("Service was not ready: {}", e.into()))
-                    })
+                /// Get a immutable reference to the inner type `T`.
+                pub fn get_ref(&self) -> &T {
+                    &self.inner
+                }
+
+                /// Get a mutable reference to the inner type `T`.
+                pub fn get_mut(&mut self) -> &mut T {
+                    &mut self.inner
                 }
 
                 #methods
@@ -110,7 +113,9 @@ fn generate_unary(method: &Method, proto: &str, path: String) -> TokenStream {
             &mut self,
             request: impl tonic::IntoRequest<#request>,
         ) -> Result<tonic::Response<#response>, tonic::Status> {
-           self.ready().await?;
+            self.inner.ready().await.map_err(|e| {
+                        tonic::Status::new(tonic::Code::Unknown, format!("Service was not ready: {}", e.into()))
+            })?;
            let codec = tonic::codec::ProstCodec::default();
            let path = http::uri::PathAndQuery::from_static(#path);
            self.inner.unary(request.into_request(), path, codec).await
@@ -128,7 +133,9 @@ fn generate_server_streaming(method: &Method, proto: &str, path: String) -> Toke
             &mut self,
             request: impl tonic::IntoRequest<#request>,
         ) -> Result<tonic::Response<tonic::codec::Streaming<#response>>, tonic::Status> {
-           self.ready().await?;
+            self.inner.ready().await.map_err(|e| {
+                        tonic::Status::new(tonic::Code::Unknown, format!("Service was not ready: {}", e.into()))
+            })?;
            let codec = tonic::codec::ProstCodec::default();
            let path = http::uri::PathAndQuery::from_static(#path);
            self.inner.server_streaming(request.into_request(), path, codec).await
@@ -146,10 +153,12 @@ fn generate_client_streaming(method: &Method, proto: &str, path: String) -> Toke
             &mut self,
             request: impl tonic::IntoStreamingRequest<Message = #request>
         ) -> Result<tonic::Response<#response>, tonic::Status> {
-           self.ready().await?;
-           let codec = tonic::codec::ProstCodec::default();
-           let path = http::uri::PathAndQuery::from_static(#path);
-           self.inner.client_streaming(request.into_streaming_request(), path, codec).await
+            self.inner.ready().await.map_err(|e| {
+                        tonic::Status::new(tonic::Code::Unknown, format!("Service was not ready: {}", e.into()))
+            })?;
+            let codec = tonic::codec::ProstCodec::default();
+            let path = http::uri::PathAndQuery::from_static(#path);
+            self.inner.client_streaming(request.into_streaming_request(), path, codec).await
         }
     }
 }
@@ -164,7 +173,9 @@ fn generate_streaming(method: &Method, proto: &str, path: String) -> TokenStream
             &mut self,
             request: impl tonic::IntoStreamingRequest<Message = #request>
         ) -> Result<tonic::Response<tonic::codec::Streaming<#response>>, tonic::Status> {
-           self.ready().await?;
+            self.inner.ready().await.map_err(|e| {
+                        tonic::Status::new(tonic::Code::Unknown, format!("Service was not ready: {}", e.into()))
+            })?
            let codec = tonic::codec::ProstCodec::default();
            let path = http::uri::PathAndQuery::from_static(#path);
            self.inner.streaming(request.into_streaming_request(), path, codec).await

--- a/tonic-build/src/client.rs
+++ b/tonic-build/src/client.rs
@@ -34,16 +34,6 @@ pub(crate) fn generate(service: &Service, proto: &str) -> TokenStream {
                     Self { inner }
                 }
 
-                /// Get a immutable reference to the inner type `T`.
-                pub fn get_ref(&self) -> &T {
-                    &self.inner
-                }
-
-                /// Get a mutable reference to the inner type `T`.
-                pub fn get_mut(&mut self) -> &mut T {
-                    &mut self.inner
-                }
-
                 #methods
             }
 
@@ -175,7 +165,7 @@ fn generate_streaming(method: &Method, proto: &str, path: String) -> TokenStream
         ) -> Result<tonic::Response<tonic::codec::Streaming<#response>>, tonic::Status> {
             self.inner.ready().await.map_err(|e| {
                         tonic::Status::new(tonic::Code::Unknown, format!("Service was not ready: {}", e.into()))
-            })?
+            })?;
            let codec = tonic::codec::ProstCodec::default();
            let path = http::uri::PathAndQuery::from_static(#path);
            self.inner.streaming(request.into_streaming_request(), path, codec).await

